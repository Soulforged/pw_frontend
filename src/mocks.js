//@flow

//FIXME remove all mocks from here when they're no longer necessary
/* eslint-disable global-require,import/prefer-default-export */
export default {
  summary: require("./data/summary.json"),
  trendsByDate: require("./data/trendsByDate.json"),
  trendsByInstitution: require("./data/trendsByInstitution.json"),
  errors: require("./data/errors.json"),
  users: require("./data/users.json"),
  login: require("./data/login.json"),
  user: require("./data/user.json"),
  userCreate: require("./data/userCreate.json"),
  role: require("./data/role.json")
};

/* eslint-disable */
