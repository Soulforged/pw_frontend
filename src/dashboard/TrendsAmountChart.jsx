//@flow
import React from "react";
import { compose, branch, renderComponent } from "recompose";
import { Line as LineChart } from "react-chartjs";
import { boundLifecycle } from "src/recompose-ext";
import { Loading } from "src/components";

const options = {
  responsive: true,
  maintainAspectRatio: true,
  title: {
    display: true,
    text: "Cash in",
    fontColor: "#fff",
    fontFamily: "robotolight"
  },
  scales: {
    xAxes: [{ gridLines: { display: false, drawBorder: false } }],
    yAxes: [{
      ticks: {
        beginAtZero: true
      },
      gridLines: { borderDash: [2, 8], drawBorder: false }
    }]
  }
};
const defDatasets = [
  {
    backgroundColor: "#707ed3",
    borderColor: "#707ed3",
    data: [],
    label: "Cash-In",
    fill: false
  }, {
    backgroundColor: "#4bd1b4",
    borderColor: "#4bd1b4",
    data: [],
    label: "Cash-Out",
    fill: false
  }
];

const SpecLoading = () => <Loading style={{ height: "398px" }} loading />;

const Chart = ({ trendsByDate }: { trendsByDate: Array<Object> }) => {
  const chartData = trendsByDate.reduce(({ labels, datasets }, trend) => (
    {
      labels: [...labels, trend.date],
      datasets: [{
        ...datasets[0],
        data: [...datasets[0].data, trend.cashInAmount]
      }, {
        ...datasets[1],
        data: [...datasets[1].data, trend.cashOutAmount]
      }]
    }
  ), { labels: [], datasets: defDatasets });
  return <LineChart data={chartData} options={options} height="90px" />;
};

export default compose(
  boundLifecycle({ didMount: ({ fetchTrendsByDate }) => fetchTrendsByDate() }),
  branch(({ trendsByDate: { fetching } }) => fetching, renderComponent(SpecLoading))
)(Chart);
