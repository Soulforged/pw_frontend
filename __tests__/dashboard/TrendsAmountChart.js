//@flow
import React from "react";
import TrendsAmountChart from "src/dashboard/TrendsAmountChart";
import renderer from "react-test-renderer";
import mocks from "src/mocks";

it("matches loading snapshot", () => {
  const component = (
    <TrendsAmountChart
      fetchTrendsByDate={jest.fn()}
      trendsByDate={{ fetching: true }}
    />
  );
  const rendered = renderer.create(component).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("matches snapshot with data", () => {
  const trendsByDate = mocks.trendsByDate.results;
  const component = <TrendsAmountChart fetchTrendsByDate={jest.fn()} trendsByDate={trendsByDate} />;
  const rendered = renderer.create(component).toJSON();
  expect(rendered).toMatchSnapshot();
});
