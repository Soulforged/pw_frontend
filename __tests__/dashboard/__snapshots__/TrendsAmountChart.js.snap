// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches loading snapshot 1`] = `
<div
  id="loader"
  style={
    Object {
      "height": "398px",
    }
  }
>
  <img
    alt="loading something..."
    className="loader"
    src="loader.gif"
  />
</div>
`;

exports[`matches snapshot with data 1`] = `
<Line
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": "#707ed3",
          "borderColor": "#707ed3",
          "data": Array [
            undefined,
            undefined,
          ],
          "fill": false,
          "label": "Cash-In",
        },
        Object {
          "backgroundColor": "#4bd1b4",
          "borderColor": "#4bd1b4",
          "data": Array [
            undefined,
            undefined,
          ],
          "fill": false,
          "label": "Cash-Out",
        },
      ],
      "labels": Array [
        "2018-02-25",
        "2018-02-27",
      ],
    }
  }
  height="90px"
  options={
    Object {
      "maintainAspectRatio": true,
      "responsive": true,
      "scales": Object {
        "xAxes": Array [
          Object {
            "gridLines": Object {
              "display": false,
              "drawBorder": false,
            },
          },
        ],
        "yAxes": Array [
          Object {
            "gridLines": Object {
              "borderDash": Array [
                2,
                8,
              ],
              "drawBorder": false,
            },
            "ticks": Object {
              "beginAtZero": true,
            },
          },
        ],
      },
      "title": Object {
        "display": true,
        "fontColor": "#fff",
        "fontFamily": "robotolight",
        "text": "Cash in",
      },
    }
  }
/>
`;
